packages/flutter/lib/src/material/dialog.dart

```diff
@@ -639,10 +582,38 @@ Future<T> showDialog<T>({
   WidgetBuilder builder,
 }) {
   assert(child == null || builder == null);
-  return Navigator.of(context, rootNavigator: true).push(new _DialogRoute<T>(
-    child: child ?? new Builder(builder: builder),
-    theme: Theme.of(context, shadowThemeOnly: true),
+  return showGeneralDialog(
+    context: context,
+    pageBuilder: (BuildContext context, Animation<double> animation, Animation<double> secondaryAnimation) {
+      final ThemeData theme = Theme.of(context, shadowThemeOnly: true);
+      final Widget pageChild =  child ?? new Builder(builder: builder);
+      return new SafeArea(
+        child: new Builder(
+          builder: (BuildContext context) {
+            final Widget annotatedChild = new Semantics(
```

Comment:
This is generally true of dialogs in general. We should move this into _DialogRoute's buildPage